src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,39 @@
 'use strict';
 
+const DEFAULT_HEALTH = 100;
+const BITE_DAMAGE = 50;
+
 class Animal {
-  // write your code here
+  constructor(name, health = DEFAULT_HEALTH) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(health, name) {
+    super(health, name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (!animal.hidden && animal instanceof Herbivore) {
+      animal.health -= BITE_DAMAGE;
+    }
+
+    if (animal.health === 0) {
+      Animal.alive = Animal.alive.filter((zvir) => zvir.health > 0);
```

Comment:
[CODE PERFORMANCE] The check for `animal.health === 0` before filtering the `Animal.alive` array is unnecessary. You should remove the `if` statement and directly filter the array to remove any animals with `health` less than or equal to `0`. This will handle cases where health might be a negative number as well and is more efficient.